torch/csrc/THP.h

```diff
@@ -15,6 +15,14 @@
 #define PyLong_FromLong         PyLong_FromLongLong
 #define PyLong_AsLong           PyLong_AsLongLong
 #define PyLong_FromUnsignedLong PyLong_FromUnsignedLongLong
+
+#ifdef PyInt_Check
+#undef PyInt_FromLong
+#undef PyInt_AsLong
+
+#define PyInt_FromLong          PyLong_FromLong
+#define PyInt_AsLong            PyLong_AsLong
```

Comment:
I’m ok with limiting the scope of this change to Windows, but we should be able to return PyInts on posix platforms (there’s no reason not to)

Reply:
Actually, the `PyInt` functions are defined below, at Line 25 - 29. This one is essential for Windows, but not necessary for Unix systems.  `int64_t` is used instead of `long`. It is the same with `long long` in Windows, but `long` in Unix systems.

Reply:
The macros below do something else - if the Int functions are **undefined**, we define them to the Long functions (this is the case for Python 3). In your case, you replace them if they are **defined**, which happens only in Python 2, making it impossible for us to return objects of `int` type (everything will become `long`).